Extensions/Lighting/lightobstacleruntimebehavior.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * @memberof gdjs
+ * @class LightObstaclesManager
+ * @param {gdjs.RuntimeScene} runtimeScene
+ */
+gdjs.LightObstaclesManager = function (runtimeScene) {
+  this._obstacleRBush = new rbush(9, [
+    '.owner.getAABB().min[0]',
+    '.owner.getAABB().min[1]',
+    '.owner.getAABB().max[0]',
+    '.owner.getAABB().max[1]',
+  ]);
+};
+
+gdjs.LightObstaclesManager.getManager = function (runtimeScene) {
+  if (!runtimeScene.lightObstaclesManager) {
```

Comment:
nit: can we use `_lightObstaclesManager`?  We're extending runtimeScene which is already a "bad enough" practice by itself (though the most efficient option I think), so let's put a underscore so that someone does not come and play with our obstacle manager ;)